src/openforms/js/components/admin/forms/VariableSelection.js

```diff
@@ -30,20 +30,16 @@ const VariableSelection = ({
       const label = formDefinitionsNames[variable.formDefinition]
         ? `${formDefinitionsNames[variable.formDefinition]}: ${variable.name} (${variable.key})`
         : `${variable.name} (${variable.key})`;
-      return [variable.key, label];
+      return {value: variable.key, label};
     });
 
-  {
-    /*TODO: This should be a searchable select for when there are a billion variables -> react-select */
-  }
   return (
-    <Select
+    <SelectWithoutFormik
       id={id}
       className="form-variable-dropdown"
       name={name}
-      choices={choices}
-      allowBlank
-      onChange={onChange}
+      options={choices}
```

Comment:
How should we approach grouping the choices into optgroups by type of variable: component / user defined / static?

Reply:
Ah yes, React-select has a default solution for this. If we change the structure of the choices variable/object, it should work.

I'll make a setup

Reply:
The variableSelections are now grouped by variable source (i.e. user defined, components and static). In order of user, component, static.

The ReactSelect now uses a recursive function to define its value. So in the future more ReactSelect components can use grouped options